soccer/writers.py

```diff
@@ -15,47 +14,48 @@
 
 
 def get_writer(output):
-    if output == 'stdout':
-        return Console()
-    elif output == 'json':
-        return JSON()
-    elif output == 'csv':
-        return CSV()
-
-
-def supported_leagues(total_data, writer):
-    """Filters out scores of unsupported leagues"""
-    supported_leagues = {val: key for key, val in LEAGUE_IDS.items()}
-    get_league_id = lambda x: int(x["_links"]["soccerseason"]["href"].split("/")[-1])
-    fixtures = (fixture for fixture in total_data["fixtures"]
-                if get_league_id(fixture) in supported_leagues)
+    return globals()[output.capitalize()]()
 
-    # Sort the scores by league to make it easier to read
-    fixtures = sorted(fixtures, key=get_league_id)
-    for league, scores in groupby(fixtures, key=get_league_id):
-        writer.league_header(supported_leagues[league])
-        for score in scores:
-            yield supported_leagues[league].strip(), score
+class BaseWriter(object):
 
+    __metaclass__ = ABCMeta
 
-class BaseWriter(object):
+    @abstractmethod
     def live_scores(self, live_scores):
         pass
 
+    @abstractmethod
     def team_scores(self, team_scores, time):
         pass
 
+    @abstractmethod
     def standings(self, league_table, league):
         pass
 
+    @abstractmethod
     def league_scores(self, total_data, time):
         pass
 
-    def league_header(self, league_name):
-        pass
+    def supported_leagues(self, total_data, stdout=True):
+        """Filters out scores of unsupported leagues"""
+        supported_leagues = {val: key for key, val in LEAGUE_IDS.items()}
+        get_league_id = lambda x: int(x["_links"]["soccerseason"]["href"].split("/")[-1])
+        fixtures = (fixture for fixture in total_data["fixtures"]
+                    if get_league_id(fixture) in supported_leagues)
+
+        # Sort the scores by league to make it easier to read
+        fixtures = sorted(fixtures, key=get_league_id)
+        for league, scores in groupby(fixtures, key=get_league_id):
+            league = supported_leagues[league]
+            if stdout:
+                league_name = " {0} ".format(league)
+                click.secho("{:=^56}".format(league_name), fg="green")
+                click.echo()
```

Comment:
I agree that it shouldn't be added to the `BaseWriter` since it only makes sense for a subset of classes. However, right now we are hard coding the code for a `Stdout` class inside the `BaseWriter` which just doesn't seem like a good idea to me.

I think it might be better if we just have it call a method that gets implemented by whatever class wants to print out the header. That code only gets reached by passing in an explicit flag, so whomever is calling it should know to provide the method to write the header.